src/libraries/System.Net.Security/src/System/Net/Security/TlsFrameHelper.cs

```diff
@@ -179,27 +179,51 @@ public static bool TryGetFrameHeader(ReadOnlySpan<byte> frame, ref TlsFrameHeade
             {
                 header.Type = (TlsContentType)frame[0];
 
-                if (frame.Length >= 3)
+                if (frame.Length > 4)
```

Comment:
Is `result` above ever reassigned?  I'm not clear why we're storing `frame.Length > 4` into it, never overwriting `result` with something else, and then returning `result` at the bottom.  Should that return at the bottom instead be `return false;` and then have a `return true;` at the end of this if block (and then delete the other `return true;` lines in some of the sub-if blocks)?

Reply:
we could fall-through here. In Ssl3 and above we should always succeed if we have at least 5 bytes. For Ssl2/Unified, we care only about ClientHello and we should never see anything else.

Reply:
I reworked the while structure to avoid some of the nesting. e.g. length check at begging with immediate false return if insufficient data. Rest of the logic follow.